sys/net/application_layer/nanocoap/nanocoap.c

```diff
@@ -922,6 +922,18 @@ ssize_t coap_opt_finish(coap_pkt_t *pkt, uint16_t flags)
     return pkt->payload - (uint8_t *)pkt->hdr;
 }
 
+ssize_t coap_payload_put_bytes(coap_pkt_t *pkt, const void *data, size_t len)
+{
+    if (pkt->payload_len < len) {
+        return -ENOMEM;
```

Comment:
I favor use of -ENOSPC for consistency with other nanocoap use. The model here is that the packet (PDU really) more closely represents a fixed size structure, rather than a memory allocation. I don't feel really strongly about this though, as the most important feature to the user is that it's <0 as you have documented it, which is what I expect they will check for.